cli/remotebazel/remotebazel.go

```diff
@@ -703,9 +703,14 @@ func Run(ctx context.Context, opts RunOpts, repoConfig *RepoConfig) (int, error)
 		reqArch = *execArch
 	}
 
+	bazelArgs := opts.Args
+	if !*runRemotely {
+		// If we are running the target locally, we append the exec arguments
+		// when we actually run it
+		bazelArgs = arg.GetBazelArgs(opts.Args)
+	}
```

Comment:
We previously did not handle those cases correctly, but with this change we will

Reply:
Well actually for the first case `bazel run -- //executable arg1 arg2` it will not work if you want to build the binary remotely and run it locally (both previously and with this PR). But if people face that, we can just ask them to move the executable before `--`. This will keep parsing much simpler